From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Oleg <balbes-150@yandex.ru>
Date: Wed, 4 May 2022 16:16:40 +0200
Subject: [ARCHEOLOGY] move 5.18-rc5 support jetson-nano (#3754)

> X-Git-Archeology: - Revision 0f7606e653c607da7b40e855098b5fd23fe9eaec: https://github.com/armbian/build/commit/0f7606e653c607da7b40e855098b5fd23fe9eaec
> X-Git-Archeology:   Date: Wed, 04 May 2022 16:16:40 +0200
> X-Git-Archeology:   From: Oleg <balbes-150@yandex.ru>
> X-Git-Archeology:   Subject: move 5.18-rc5 support jetson-nano (#3754)
> X-Git-Archeology:
> X-Git-Archeology: - Revision ba7912c0c9f5267aa19438117c265c6f7f635ef2: https://github.com/armbian/build/commit/ba7912c0c9f5267aa19438117c265c6f7f635ef2
> X-Git-Archeology:   Date: Mon, 23 May 2022 16:17:09 +0200
> X-Git-Archeology:   From: Oleg <balbes-150@yandex.ru>
> X-Git-Archeology:   Subject: move kernel media 5.17.9 and 5.18-rc7 (#3792)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 2e1b4eed739d6ea81c3a7ba7e94d1bcab5cbbb6c: https://github.com/armbian/build/commit/2e1b4eed739d6ea81c3a7ba7e94d1bcab5cbbb6c
> X-Git-Archeology:   Date: Mon, 04 Jul 2022 16:18:31 +0300
> X-Git-Archeology:   From: balbes150 <balbes-150@yandex.ru>
> X-Git-Archeology:   Subject: move kernel media to current 5.18 and edge 5.19
> X-Git-Archeology:
> X-Git-Archeology: - Revision 3ff351cc52aed68e80949576f43d2fdf08ab8b5f: https://github.com/armbian/build/commit/3ff351cc52aed68e80949576f43d2fdf08ab8b5f
> X-Git-Archeology:   Date: Thu, 06 Oct 2022 18:07:36 +0200
> X-Git-Archeology:   From: Oleg <balbes-150@yandex.ru>
> X-Git-Archeology:   Subject: move media current 5.19 and edge 6.0 (#4262)
> X-Git-Archeology:
> X-Git-Archeology: - Revision c29e51c70828145785c3167d984e6857e342c96e: https://github.com/armbian/build/commit/c29e51c70828145785c3167d984e6857e342c96e
> X-Git-Archeology:   Date: Sat, 12 Nov 2022 10:51:25 +0100
> X-Git-Archeology:   From: Oleg <balbes-150@yandex.ru>
> X-Git-Archeology:   Subject: move kernel media-edge to 6.0.7 (#4405)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 276f346ee9c21cdcd627e54676334f4d5a9292b6: https://github.com/armbian/build/commit/276f346ee9c21cdcd627e54676334f4d5a9292b6
> X-Git-Archeology:   Date: Wed, 14 Dec 2022 10:17:09 +0100
> X-Git-Archeology:   From: Oleg <balbes-150@yandex.ru>
> X-Git-Archeology:   Subject: move kernel media to current 6.0 and edge 6.1 (#4566)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 462b3befec60cfe998b67de82dc62836c0376ef2: https://github.com/armbian/build/commit/462b3befec60cfe998b67de82dc62836c0376ef2
> X-Git-Archeology:   Date: Thu, 26 Jan 2023 19:08:48 +0100
> X-Git-Archeology:   From: Oleg <balbes-150@yandex.ru>
> X-Git-Archeology:   Subject: correction of bpir2pro frequency control and changing the composition of images (#4753)
> X-Git-Archeology:
---
 arch/arm64/boot/dts/rockchip/rk3568-bpi-r2-pro.dts | 124 +++++++++-
 1 file changed, 122 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3568-bpi-r2-pro.dts b/arch/arm64/boot/dts/rockchip/rk3568-bpi-r2-pro.dts
index 26d7fda275ed..5ff9898b5fec 100644
--- a/arch/arm64/boot/dts/rockchip/rk3568-bpi-r2-pro.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3568-bpi-r2-pro.dts
@@ -6,6 +6,7 @@
 
 /dts-v1/;
 #include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/pwm/pwm.h>
 #include <dt-bindings/leds/common.h>
 #include <dt-bindings/pinctrl/rockchip.h>
 #include <dt-bindings/soc/rockchip,vop2.h>
@@ -55,6 +56,14 @@ dc_12v: dc-12v-regulator {
 		regulator-max-microvolt = <12000000>;
 	};
 
+	fan: gpio_fan {
+		compatible = "gpio-fan";
+		gpios = <&gpio0 RK_PD5 GPIO_ACTIVE_HIGH>;
+		gpio-fan,speed-map = <0    0
+				      4500 1>;
+		#cooling-cells = <2>;
+	};
+
 	hdmi-con {
 		compatible = "hdmi-connector";
 		type = "a";
@@ -179,6 +188,28 @@ vcc5v0_usb_otg: vcc5v0-usb-otg-regulator {
 		regulator-max-microvolt = <5000000>;
 		vin-supply = <&vcc5v0_usb>;
 	};
+
+	rk809-sound {
+		compatible = "simple-audio-card";
+		simple-audio-card,format = "i2s";
+		simple-audio-card,name = "Analog RK809";
+		simple-audio-card,mclk-fs = <256>;
+
+		simple-audio-card,cpu {
+			sound-dai = <&i2s1_8ch>;
+		};
+		simple-audio-card,codec {
+			sound-dai = <&rk809>;
+		};
+	};
+
+	rk_headset: rk-headset {
+		compatible = "rockchip_headset";
+		headset_gpio = <&gpio3 RK_PC2 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&hp_det>;
+		io-channels = <&saradc 2>;    //HP_HOOK pin
+	};
 };
 
 &combphy0 {
@@ -194,6 +225,39 @@ &combphy1 {
 &combphy2 {
 	/* used for SATA */
 	status = "okay";
+//};
+
+//&cpu0 {
+//	cpu-supply = <&vdd_cpu>;
+//};
+
+//&cpu1 {
+//	cpu-supply = <&vdd_cpu>;
+//};
+
+//&cpu2 {
+//	cpu-supply = <&vdd_cpu>;
+//};
+
+//&cpu3 {
+//	cpu-supply = <&vdd_cpu>;
+//};
+
+//&cpu_thermal {
+//	trips {
+//		cpu_hot: cpu_hot {
+//			temperature = <55000>;
+//			hysteresis = <2000>;
+//			type = "active";
+//		};
+//	};
+
+//	cooling-maps {
+//		map1 {
+//			trip = <&cpu_hot>;
+//			cooling-device = <&fan THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+//		};
+//	};
 };
 
 &gmac0 {
@@ -276,15 +340,49 @@ &hdmi_sound {
 &i2c0 {
 	status = "okay";
 
+
+//	vdd_cpu: regulator@1c {
+//		compatible = "tcs,tcs4525";
+//		reg = <0x1c>;
+//		vin-supply = <&vcc5v0_sys>;
+//		regulator-compatible = "fan53555-reg";
+//		regulator-name = "vdd_cpu";
+//		regulator-min-microvolt = <712500>;
+//		regulator-max-microvolt = <1390000>;
+//		regulator-ramp-delay = <2300>;
+//		fcs,suspend-voltage-selector = <1>;
+//		regulator-boot-on;
+//		regulator-always-on;
+
+//		regulator-state-mem {
+//			regulator-off-in-suspend;
+//		};
+//	};
+
 	rk809: pmic@20 {
 		compatible = "rockchip,rk809";
 		reg = <0x20>;
 		interrupt-parent = <&gpio0>;
 		interrupts = <RK_PA3 IRQ_TYPE_LEVEL_LOW>;
+		assigned-clocks = <&cru I2S1_MCLKOUT>, <&cru I2S1_MCLK_TX_IOE>;
+		assigned-clock-rates = <12288000>;
+		assigned-clock-parents = <&cru I2S1_MCLKOUT_TX>, <&cru I2S1_MCLKOUT_TX>;
 		#clock-cells = <1>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&pmic_int>;
+		clock-names = "mclk";
+		clocks = <&cru I2S1_MCLKOUT>;
+		pinctrl-names = "default", "pmic-sleep",
+				"pmic-power-off", "pmic-reset";
+		pinctrl-0 = <&pmic_int>, <&i2s1m0_mclk>;
+
 		rockchip,system-power-controller;
+		#sound-dai-cells = <0>;
+		clock-output-names = "rk808-clkout1", "rk808-clkout2";
+		//fb-inner-reg-idxs = <2>;
+		/* 1: rst regs (default in codes), 0: rst the pmic */
+		pmic-reset-func = <0>;
+		/* not save the PMIC_POWER_EN register in uboot */
+		not-save-power-en = <1>;
+
 		vcc1-supply = <&vcc3v3_sys>;
 		vcc2-supply = <&vcc3v3_sys>;
 		vcc3-supply = <&vcc3v3_sys>;
@@ -488,6 +586,10 @@ regulator-state-mem {
 				};
 			};
 		};
+
+		codec {
+			mic-in-differential;
+		};
 	};
 };
 
@@ -517,6 +619,18 @@ &i2s0_8ch {
 	status = "okay";
 };
 
+&i2s1_8ch {
+	/* headphone */
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2s1m0_sclktx
+		     &i2s1m0_lrcktx
+		     &i2s1m0_sdi0
+		     &i2s1m0_sdo0>;
+	rockchip,trcm-sync-tx-only;
+	status = "okay";
+};
+
+
 &mdio0 {
 	#address-cells = <1>;
 	#size-cells = <0>;
@@ -648,6 +762,12 @@ vcc5v0_usb_otg_en: vcc5v0_usb_otg_en {
 			rockchip,pins = <0 RK_PA5 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
+
+	headphone {
+		hp_det: hp-det {
+			rockchip,pins = <3 RK_PC2 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+	};
 };
 
 &pmu_io_domains {
-- 
Armbian

